### ISSUE TITLE
it is not changing to GPU even though I am using device = 'gpu'

### ISSUE THREAD

**Issue by vivek-ram** [apple/tensorflow_macos#71]
from tensorflow.python.compiler.mlcompute import mlcompute
mlcompute.set_mlc_device(device_name='gpu')

WARNING:tensorflow:Eager mode uses the CPU. Switching to the CPU.

**Comment by jzinnegger** [comment]
On my MacBook Pro 2015 I have the same behaviour. However, when graph mode is activated, e.g. while running model.fit(), I can observe that the activity monitor indicates that the corresponding python process utilises the GPU at 100%. It falls to 0% after completion of model.fit().

<img width="721" alt="Screenshot 2020-12-14 at 19 23 15" src="https://user-images.githubusercontent.com/54271028/102119929-22f24b00-3e42-11eb-8a71-ae9ece2a02ca.png">

**Comment by vivek-ram** [comment]
oh yeah but when I run code in gpu yeah it slows my MacBook pro so much is that normal ? or should I only specify use only particular number of cores ?? I am using 2020 MacBook Pro m1

**Comment by anna-tikhonova** [comment]
> from tensorflow.python.compiler.mlcompute import mlcompute
> mlcompute.set_mlc_device(device_name='gpu')
> 
> WARNING:tensorflow:Eager mode uses the CPU. Switching to the CPU.

This is expected behavior. In eager mode, we use the CPU.

**Comment by lamerman** [comment]
@anna-tikhonova I explicitly set eager mode to False in Keras

`model.compile(optimizer='adam',
                  loss=tf.keras.losses.SparseCategoricalCrossentropy(from_logits=True),
                  metrics=['accuracy'],
                  run_eagerly=False)`

Tensorflow tells me `WARNING:tensorflow:Eager mode uses the CPU. Switching to the CPU.`

But the activity monitor show that GPU is being in use (95%+). In contrast to the CPU mode, where GPU is not loaded and CPU is loaded at 170%. 
And the execution time differs when I set CPU and GPU on apple m1. For GPU it's 42 seconds for epoch, for CPU 70 seconds.

It looks like it actually runs on GPU but mistakenly says that switches to CPU. But not sure.

**Comment by rkraghu88** [comment]
I am facing some issues too, with mlcompute in my M1 Macbook Air. In my case it's the other way around. Whatever I set the mlcompute to ('gpu/cpu/any'). it always uses 85% GPU.
